core/ai.go

```diff
@@ -127,3 +133,122 @@ func ParseStepsFromModelID(modelID *string, defaultSteps float64) float64 {
 
 	return numInferenceSteps
 }
+
+type ModelsWebhook struct {
+	configs    []AIModelConfig
+	source     string
+	mu         sync.RWMutex
+	lastHash   string
+	refreshInt time.Duration
+	stopChan   chan struct{}
+}
+
+func NewAIModelWebhook(source string, refreshInterval time.Duration) (*ModelsWebhook, error) {
+	webhook := &ModelsWebhook{
+		source:     source,
+		refreshInt: refreshInterval,
+		stopChan:   make(chan struct{}),
+	}
+	err := webhook.refreshConfigs()
+	if err != nil {
+		return nil, err
+	}
+	go webhook.startRefreshing()
+	return webhook, nil
+}
+
+func FetchAIModelConfigs(source string) ([]AIModelConfig, error) {
```

Comment:
`FetchAIModelConfigs()` looks almost the same like `fetchContent()`. Could we reuse it somehow?

Reply:
We can remove it https://github.com/livepeer/go-livepeer/pull/3209#discussion_r1803425990